subprojects/core/src/main/java/org/gradle/internal/buildevents/BuildExceptionReporter.java

```diff
@@ -66,29 +68,38 @@ public void buildFinished(BuildResult result) {
             return;
         }
 
-        execute(failure);
+        execute(failure, isGradleEnterprisePluginApplied(result));
     }
 
     @Override
     public void execute(Throwable failure) {
+        execute(failure, false);
+    }
+
+    private void execute(Throwable failure, boolean gradleEnterprisePluginApplied) {
         if (failure instanceof MultipleBuildFailures) {
             List<? extends Throwable> flattenedFailures = ((MultipleBuildFailures) failure).getCauses();
-            renderMultipleBuildExceptions(failure.getMessage(), flattenedFailures);
+            renderMultipleBuildExceptions(failure.getMessage(), flattenedFailures, gradleEnterprisePluginApplied);
             return;
         }
 
-        renderSingleBuildException(failure);
+        renderSingleBuildException(failure, gradleEnterprisePluginApplied);
+    }
+
+    private boolean isGradleEnterprisePluginApplied(BuildResult result) {
+        GradleInternal gradle = (GradleInternal) result.getGradle();
+        return gradle != null && gradle.getSettings().getPluginManager().hasPlugin(AutoAppliedGradleEnterprisePlugin.ID.getId());
```

Comment:
Probably better to use `GradleEnterprisePluginManager.isPresent()` instead, to be more consistent with other sites.

Reply:
Ah, didn't even see this there.  And the other approach was causing problems under certain conditions, as well.  Changed.